src/WebJobs.Script/Workers/SharedMemoryDataTransfer/SharedMemoryMap.cs

```diff
@@ -0,0 +1,296 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer
+{
+    /// <summary>
+    /// TODO:
+    /// 1) Add class description.
+    /// 2) Logs to LogDebug or LogTrace.
+    /// 3) Make sure all meaningful log places are logging.
+    /// 4) long and int consistency - ensure > 2GB mmaps are handled.
+    /// 5) put all private methods at the end.
+    /// </summary>
+    public class SharedMemoryMap : IDisposable
+    {
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// <see cref="MemoryMappedFile"/> containg the header + content of this <see cref="SharedMemoryMap"/>.
+        /// </summary>
+        private readonly MemoryMappedFile _mmf;
+
+        /// <summary>
+        /// Name of the <see cref="MemoryMappedFile"/> backing this <see cref="SharedMemoryMap"/>.
+        /// </summary>
+        private readonly string _mapName;
+
+        private readonly IMemoryMappedFileAccessor _mapAccessor;
+
+        public SharedMemoryMap(ILogger logger, IMemoryMappedFileAccessor mapAccessor, string mapName, MemoryMappedFile mmf)
```

Comment:
Instead of logger, pass `LoggerFactory` and create logger of this type : `LoggerFactory.Create<SharedMemoryMap>()`